java/client/src/org/openqa/selenium/WebDriver.java

```diff
@@ -91,11 +91,19 @@
    * found collection, or will return an empty list if the timeout is reached.
    *
    * @param by The locating mechanism to use
+   * 
    * @return A list of all {@link WebElement}s, or an empty list if nothing matches
+   * 
+   * @throws ClassCastException The WebDriver interface is generic. 
```

Comment:
@alexec this is just information. If end users will re-implement the proposed API then there won't be any problem. An end user will face a problem when they use common Selenium implementations, because it not so flexible like API.